### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by Draxeytina**
- [x] Create a class Classroom with the following:

    @label instance variable, should be initialized in the constructor.
    Setter and getter for @label (remember about attr_accessor).
    Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
    Create the has-many side (a classroom has many students).
    Create the belongs-to side (a student belongs to a classroom).
    Make sure that when adding a student to a classroom it also sets the classroom for the student.
    Make sure that when setting the classroom for a student it also adds it to the classrooms' students.

- [x] Create a class Book with the following:

    @title and @author instance variables, should be initialized in the constructor.
    Setters and getters for instance variables (remember about attr_accessor).

- [x] Create a class Rental with the following:

    @date instance variable, should be initialized in the constructor.
    Setter and getter for @date (remember about attr_accessor).

- [x] Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:

    Create the has-many side of Book and Rental (a book has many rentals).
    Create the belongs-to side of Rental and Book (a rental belongs to a book).
    Create the has-many side of Person and Rental (a person has many rentals).
    Create the belongs-to side of Rental and Person (a rental belongs to a person).
    Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: classes/classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  attr_accessor :label
+  attr_reader :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @students << student unless @students.include?(student)
```
Comment (omarsalem7) [inline classes/classroom.rb:11]: - [x] Kindly add a relation to classroom (classroom has many students) by adding this line `student.classroom = self`

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @Draxeytina 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- The Pull Request has a proper title and description. ✔️
- Correct git-flow ✔️

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/classroom.rb (hunk 1 above)
Comment (Draxeytina) [reply classes/classroom.rb:11]: Oh thank you @omarsalem7 let me implement this

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @Draxeytina, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You did well to make the changes requested by the previous reviewer✔️✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Draxeytina** [comment]
> Hi 👋 @Draxeytina,
> 
> Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉
> 
> ## Highlights
> * You did well to make the changes requested by the previous reviewer✔️✔️
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

thank you @elisha2kyakpo1 !

**Pull request merged**